src/sentry/api/endpoints/organization_pinned_searches.py

```diff
@@ -46,7 +46,10 @@ def put(self, request: Request, organization) -> Response:
             values={"query": result["query"], "sort": result["sort"]},
         )
         pinned_search = SavedSearch.objects.get(
-            organization=organization, owner=request.user, type=result["type"]
+            organization=organization,
+            owner=request.user,
+            type=result["type"],
+            visibility=Visibility.OWNER_PINNED,
```

Comment:
We specifically check this conditional now so that we can allow a single owner to have multiple saved searches.